index.js

```diff
@@ -21,30 +22,20 @@ const exec = (cmd, args) => {
 	).filter(Boolean);
 };
 
-const install = (bin, useYarn) => () => execa(bin, ['install'])
-	.catch(err => {
-		if (err.code === 'ENOENT' && useYarn) {
-			return execa('npm', ['install']);
-		}
-		throw err;
-	});
-
 module.exports = (input, opts) => {
 	input = input || 'patch';
 
 	opts = Object.assign({
 		cleanup: true,
-		publish: true
-	}, opts, {
-		yarn: opts && opts.yarn !== false
-	});
+		publish: true,
+		yarn: hasYarn()
+	}, opts);
```

Comment:
Oh ok, I see the problem you are describing, I thought it did it in a different order.

I would propose adding something like the following for greater transparency.

```js
if (!hasYarn() && opts.yarn){
  throw new Error('Tried to use Yarn without a yarn.lock file.');
}
```

Do you agree?